lib/ActivityStreams.js

```diff
@@ -164,16 +173,69 @@ ActivityStreams.prototype = {
    * Sets up the toolbar button that links to the Timeline page.
    */
   _setupButton() {
-    ActionButton({
+    this._button = ActionButton({
       id: "activity-streams-link",
       label: "Timeline",
       icon: data.url("content/img/list-icon.svg"),
       // TODO: Fix url below to about:timeline when that gets implemented.
-      onClick: () => tabs.open("resource://activity-streams/data/content/activity-streams.html#timeline")
+      onClick: () => tabs.open(`${this.options.pageURL}#/timeline`)
     });
   },
 
   /**
+   * The URLs for the app.
+   */
+  get appURLs() {
+    let baseUrl = this.options.pageURL;
+    return [
+      baseUrl,
+      `${baseUrl}#/`,
+      `${baseUrl}#/timeline`,
+    ];
+  },
+
+  /**
+   * Add the initial pages to the passed in window.
+   */
+  _addInitialPages(window) {
+    if (window.gInitialPages) {
+      for (let url of this.appURLs) {
+        window.gInitialPages.push(url);
+      }
+    }
+  },
+
+  /**
+   * Watches for new windows and adds our urls to gInitialPages.
+   */
+  _windowObserver: {
+    observe: (chromeWindow, topic) => {
+      if (topic === "domwindowopened") {
+        let window = chromeWindow;
+        window.QueryInterface(Ci.nsIDOMWindow).addEventListener("DOMContentLoaded", {
+          handleEvent: () => {
```

Comment:
Not sure I follow? We need to have it set on every open window in case they open a new tab.